BraveWallet/Crypto/Stores/SendTokenStore.swift

```diff
@@ -233,42 +267,89 @@ public class SendTokenStore: ObservableObject {
+            completion(success, errorMessage)
+          }
+        } else {
+          let txDataUnion = BraveWallet.TxDataUnion(ethTxData: baseData)
+          self.txService.addUnapprovedTransaction(txDataUnion, from: fromAddress, origin: nil) { success, txMetaId, errorMessage in
+            self.isMakingTx = false
+            completion(success, errorMessage)
+          }
+        }
+      } else {
+        self.ethTxManagerProxy.makeErc20TransferData(self.sendAddress, amount: "0x\(weiHexString)") { success, data in
+          guard success else {
+            completion(false, nil)
+            return
+          }
+          let baseData = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: token.contractAddress, value: "0x0", data: data)
           if network.isEip1559 {
-            self.makeEIP1559Tx(chainId: network.chainId, baseData: baseData, from: fromAddress) { success in
+            self.makeEIP1559Tx(chainId: network.chainId, baseData: baseData, from: fromAddress) { success, errorMessage  in
               self.isMakingTx = false
-              completion(success)
+              completion(success, errorMessage)
             }
           } else {
             let txDataUnion = BraveWallet.TxDataUnion(ethTxData: baseData)
             self.txService.addUnapprovedTransaction(txDataUnion, from: fromAddress, origin: nil) { success, txMetaId, errorMessage in
               self.isMakingTx = false
-              completion(success)
+              completion(success, errorMessage)
             }
           }
-        } else {
-          self.ethTxManagerProxy.makeErc20TransferData(self.sendAddress, amount: "0x\(weiHexString)") { success, data in
-            guard success else {
-              completion(false)
-              return
-            }
-            let baseData = BraveWallet.TxData(nonce: "", gasPrice: "", gasLimit: "", to: token.contractAddress, value: "0x0", data: data)
-            if network.isEip1559 {
-              self.makeEIP1559Tx(chainId: network.chainId, baseData: baseData, from: fromAddress) { success in
-                self.isMakingTx = false
-                completion(success)
-              }
-            } else {
-              let txDataUnion = BraveWallet.TxDataUnion(ethTxData: baseData)
-              self.txService.addUnapprovedTransaction(txDataUnion, from: fromAddress, origin: nil) { success, txMetaId, errorMessage in
-                self.isMakingTx = false
-                completion(success)
-              }
-            }
+        }
+      }
+    }
+  }
+  
+  private func sendTokenOnSol(
+    amount: String,
+    completion: @escaping (_ success: Bool, _ errMsg: String?) -> Void
+  ) {
+    guard let token = selectedSendToken,
+          let fromAddress = currentAccountAddress,
+          let lamports = WeiFormatter.decimalToLamports(amount)
+    else {
+      completion(false, "An Internal Error")
+      return
+    }
+    
+    rpcService.network(.eth) { [weak self] network in
```

Comment:
`sendTokenOnSol` but fetching network for `.eth` coin type? Getting `"An internal error has occurred"` error message from `addUnapprovedTransaction` when trying to send SOL.